tests/tests.go

```diff
@@ -749,6 +776,26 @@ func (ctl *controller) approveIssueNext(issue *api.Issue) error {
 	return nil
 }
 
+// approveIssueTasksWithStageApproval approves all pending approval tasks in the next stage.
+func (ctl *controller) approveIssueTasksWithStageApproval(issue *api.Issue) error {
+	for _, stage := range issue.Pipeline.StageList {
+		for _, task := range stage.TaskList {
+			if task.Status == api.TaskPendingApproval {
+				if _, err := ctl.patchStageTasksStatus(
+					api.StageTasksStatusPatch{
+						ID:     stage.ID,
+						Status: api.TaskPending,
+					},
+					issue.Pipeline.ID); err != nil {
+					return fmt.Errorf("failed to patch task status for stage %d, error: %w", stage.ID, err)
+				}
+				return nil
```

Comment:
This seems to exit after approving the first task instead of all tasks in the first stage.

Reply:
Done. Was trying to save some control logic because of the nested for-loop.